### PR TITLE
add: validation, comment-writing and admin-actions

### PR DESCRIPTION
**Pull request by chase-moskal**
add: validation

validation for comment-writing-service and admin-actions-service

### CODE REVIEW

File: s/api/services/admin-actions-service.ts [hunk 1]
```diff
@@ -3,19 +3,26 @@ import * as dbmage from "dbmage"
 import * as renraku from "renraku"
 
 import {Auth} from "../types/auth.js"
+import {BanUser, UnBanUser} from "../types/concepts.js"
+import {enforceValidation} from "./utils/enforce-validation.js"
+import {validateAdminBanUser} from "./validators/validateAdminBanUser"
```
Comment (chase-moskal) [inline s/api/services/admin-actions-service.ts:8]: all 'direct' es module imports must include `.js` extensions -- these will throw errors

please add .js on all imports of other files

File: s/api/services/admin-actions-service.ts [hunk 2]
```diff
@@ -3,19 +3,26 @@ import * as dbmage from "dbmage"
...
 export const makeAdminActionsService = () => ({
 		user, rando, database,
 	}: Auth) => ({
 
-	async banUser({userId, until}: {
-			userId: string
-			until: number
-		}): Promise<void> {
+	async banUser(rawData: BanUser): Promise<void> {
+		const data = enforceValidation(rawData, validateAdminBanUser)
```
Comment (chase-moskal) [inline s/api/services/admin-actions-service.ts:16]: we can reduce these two lines down to a single line:

```ts
const {userId, until} = enforceValidation(rawData, validateAdminBanUser)
```

File: s/api/services/comment-writing-service.ts [hunk 3]
```diff
@@ -7,7 +7,8 @@ import {Auth} from "../types/auth.js"
 import {rowToComment} from "./utils/row-to-comment.js"
 import {newCommentRow} from "./utils/new-comment-row.js"
 import {enforceValidation} from "./utils/enforce-validation.js"
-import {validateCommentPostDraft} from "./validators/validators.js"
+import {validateCommentPostDraft} from "./validators/validateCommentPostDraft"
```
Comment (chase-moskal) [inline s/api/services/comment-writing-service.ts:10]: also .js file extension are required

File: s/api/services/comment-writing-service.ts [hunk 4]
```diff
@@ -34,8 +35,10 @@ export const makeCommentWritingService = () => ({
 		return rowToComment(newRow)
 	},
 
-	async editComment(draft: CommentEditDraft): Promise<void> {
+	async editComment(rawDraft: CommentEditDraft): Promise<void> {
+		const draft = enforceValidation(rawDraft, validateCommentEditDraft)
```
Comment (chase-moskal) [inline s/api/services/comment-writing-service.ts:39]: also with the reducing 2 lines into 1

File: s/api/services/comment-writing-service.ts [hunk 5]
```diff
@@ -60,4 +63,4 @@ export const makeCommentWritingService = () => ({
 	async archiveComment(id: string): Promise<void> {
 		throw new Error("todo implement")
 	},
-})
+})
```
Comment (chase-moskal) [inline s/api/services/comment-writing-service.ts:66]: no newline at end of files, there should be

File: s/api/types/concepts.ts [hunk 6]
```diff
@@ -1,34 +1,43 @@
...
 export interface CommentPost {
-	id: string
-	parentCommentId: undefined | string
-	authorId: string
-	topicId: string
-	timePosted: number
-	subject: string
-	body: string
-	rating?: number
+  id: string;
```
Comment (chase-moskal) [inline s/api/types/concepts.ts:3]: - indentation here is incorrect, must use tabs
- we don't use semicolons in this project, please remove them

File: s/api/types/concepts.ts [hunk 7]
```diff
@@ -1,34 +1,43 @@
...
+  rating?: number;
 }
 
 export interface CommentPostDraft {
-	parentCommentId: undefined | string
-	topicId: string
-	subject: string
-	body: string
-	rating?: number
+  parentCommentId: undefined | string;
```
Comment (chase-moskal) [inline s/api/types/concepts.ts:14]: semicolons and whitespace issues throughout this file

**Review by chase-moskal — COMMENTED** [review]
some good progress here, but there are some code quality issues we need to fix.

are you using vscode? the vscode settings should have prevented several of these problems

**Review by PiusLucky — APPROVED** [review]
Everything seems to be in order.

**Comment by chase-moskal** [comment]
defunct in favor if new pr #40

**Pull request closed without merging**